feature_flag_test.go

```diff
@@ -706,3 +708,71 @@ func Test_UseCustomBucketingKey(t *testing.T) {
 		assert.Equal(t, want, got)
 	}
 }
+
+func Test_DisableNotifierOnInit(t *testing.T) {
+	tests := []struct {
+		name                 string
+		config               *ffclient.Config
+		disableNotification  bool
+		expectedNotifyCalled bool
+	}{
+		{
+			name: "DisableNotifierOnInit is true",
+			config: &ffclient.Config{
+				PollingInterval:       60 * time.Second,
+				Retriever:             &fileretriever.Retriever{Path: "testdata/flag-config.yaml"},
+				DisableNotifierOnInit: true,
+			},
+			expectedNotifyCalled: false,
+		},
+		{
+			name: "DisableNotifierOnInit is false",
+			config: &ffclient.Config{
+				PollingInterval:       60 * time.Second,
+				Retriever:             &fileretriever.Retriever{Path: "testdata/flag-config.yaml"},
+				DisableNotifierOnInit: false,
+			},
+			expectedNotifyCalled: true,
+		},
+		{
+			name: "DisableNotifierOnInit is not set",
+			config: &ffclient.Config{
+				PollingInterval: 60 * time.Second,
+				Retriever:       &fileretriever.Retriever{Path: "testdata/flag-config.yaml"},
+			},
+			expectedNotifyCalled: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockNotifier := &mockNotificationService{}
+			tt.config.Notifiers = []notifier.Notifier{mockNotifier}
+
+			gffClient, err := ffclient.New(*tt.config)
+			assert.NoError(t, err)
+			defer gffClient.Close()
+
+			time.Sleep(2 * time.Second) // wait for the goroutine to call Notify()
+			assert.Equal(t, tt.expectedNotifyCalled, mockNotifier.wasNotifyCalled())
+		})
+	}
+}
+
+type mockNotificationService struct {
+	notifyCalled bool
+	mu           sync.Mutex
+}
+
+func (m *mockNotificationService) Notify(diff notifier.DiffCache) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.notifyCalled = true
+	return nil
+}
+
+func (m *mockNotificationService) wasNotifyCalled() bool {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.notifyCalled
+}
```

Comment:
There is already a mock notifier available in the [`testutils/mock` directory](https://github.com/thomaspoignant/go-feature-flag/blob/main/testutils/mock/notifier_mock.go).

I think it would be better to use this one instead.

Reply:
Oops, I didn't notice that. Fixed!
(I've also added the missing `mutex` logic to the Notifier mock, since it seems like other mocks like `exporter_mock` have similar mutex control to avoid race condition)